commands/outbox_daemon_test.go

```diff
@@ -0,0 +1,72 @@
+package commands_test
+
+import (
+	"testing"
+
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
+
+	"github.com/filecoin-project/go-filecoin/fixtures"
+	th "github.com/filecoin-project/go-filecoin/testhelpers"
+)
+
+func TestOutbox(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	sendMessage := func(d *th.TestDaemon, from string, to string) *th.Output {
+		return d.RunSuccess("message", "send",
+			"--from", from,
+			"--gas-price", "0", "--gas-limit", "300",
+			"--value=10", to,
+		)
+	}
+
+	t.Run("list queued messages", func(t *testing.T) {
+		t.Parallel()
+		d := th.NewDaemon(t, th.KeyFile(fixtures.KeyFilePaths()[0]), th.KeyFile(fixtures.KeyFilePaths()[1])).Start()
+		defer d.ShutdownSuccess()
+
+		c1 := sendMessage(d, fixtures.TestAddresses[0], fixtures.TestAddresses[2]).ReadStdoutTrimNewlines()
+		c2 := sendMessage(d, fixtures.TestAddresses[0], fixtures.TestAddresses[2]).ReadStdoutTrimNewlines()
+		c3 := sendMessage(d, fixtures.TestAddresses[1], fixtures.TestAddresses[2]).ReadStdoutTrimNewlines()
+
+		out := d.RunSuccess("outbox", "ls").ReadStdout()
+		assert.Contains(out, fixtures.TestAddresses[0])
+		assert.Contains(out, fixtures.TestAddresses[1])
+		assert.Contains(out, c1)
+		assert.Contains(out, c2)
+		assert.Contains(out, c3)
+
+		// With address filter
+		out = d.RunSuccess("outbox", "ls", fixtures.TestAddresses[1]).ReadStdout()
+		assert.NotContains(out, fixtures.TestAddresses[0])
+		assert.Contains(out, fixtures.TestAddresses[1])
+		assert.NotContains(out, c1)
+		assert.NotContains(out, c2)
+		assert.Contains(out, c3)
+	})
+
+	t.Run("clear queue", func(t *testing.T) {
+		t.Parallel()
+		d := th.NewDaemon(t, th.KeyFile(fixtures.KeyFilePaths()[0]), th.KeyFile(fixtures.KeyFilePaths()[1])).Start()
+		defer d.ShutdownSuccess()
+
+		c1 := sendMessage(d, fixtures.TestAddresses[0], fixtures.TestAddresses[2]).ReadStdoutTrimNewlines()
+		c2 := sendMessage(d, fixtures.TestAddresses[0], fixtures.TestAddresses[2]).ReadStdoutTrimNewlines()
+		c3 := sendMessage(d, fixtures.TestAddresses[1], fixtures.TestAddresses[2]).ReadStdoutTrimNewlines()
+
+		// With address filter
+		d.RunSuccess("outbox", "clear", fixtures.TestAddresses[1])
+		out := d.RunSuccess("outbox", "ls").ReadStdout()
+		assert.Contains(out, fixtures.TestAddresses[0])
+		assert.NotContains(out, fixtures.TestAddresses[1]) // Cleared
+		assert.Contains(out, c1)
+		assert.Contains(out, c2)
+		assert.NotContains(out, c3) // cleared
+
+		// #nofilter
+		d.RunSuccess("outbox", "clear")
```

Comment:
might want to repopulate with the messages above before clearing so we know that it clears multiple addresses. also probably worth calling clear on a the empty queue afterwards so we can be sure it doesn't error.